x-pack/plugins/integration_assistant/server/util/llm.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  ActionsClientChatOpenAI,
+  ActionsClientBedrockChatModel,
+  ActionsClientSimpleChatModel,
+  ActionsClientGeminiChatModel,
+} from '@kbn/langchain/server';
+export const getLlmType = (actionTypeId: string): string | undefined => {
+  const llmTypeDictionary: Record<string, string> = {
+    [`.gen-ai`]: `openai`,
+    [`.bedrock`]: `bedrock`,
+    [`.gemini`]: `gemini`,
+  };
+  return llmTypeDictionary[actionTypeId];
+};
+
+export const getLlmClass = (llmType?: string) =>
+  llmType === 'openai'
+    ? ActionsClientChatOpenAI
+    : llmType === 'bedrock'
+    ? ActionsClientBedrockChatModel
+    : llmType === 'gemini'
+    ? ActionsClientGeminiChatModel
+    : ActionsClientSimpleChatModel;
```

Comment:
`switch` might be cleaner? And extendable too?

```ts
export const getLLMClass = (llmType?: string):ChatModels =>
    {
        switch(llmType) {
            case "openai":
                return ActionsClientChatOpenAI;
            case "bedrock":
                return ActionsClientBedrockChatModel;
            case "gemini":
                return ActionsClientGeminiChatModel;
            default:
                return ActionsClientSimpleChatModel;
        }
    };
```

Reply:
We can also extract the providers into an enum. Unless Inference does it differently altogether